homeassistant/components/solaredge/config_flow.py

```diff
@@ -66,7 +84,23 @@ async def async_step_user(self, user_input=None):
                 can_connect = await self.hass.async_add_executor_job(
                     self._check_site, site, api
                 )
+                can_connect_ha = False
+                if CONF_ACCESS_TOKEN in user_input:
+                    token = user_input[CONF_ACCESS_TOKEN]
+                    if token != "":
+                        can_connect_ha = await self.hass.async_add_executor_job(
+                            self._check_ha_site, site, token
+                        )
                 if can_connect:
+                    if can_connect_ha:
+                        return self.async_create_entry(
+                            title=name,
+                            data={
+                                CONF_SITE_ID: site,
+                                CONF_API_KEY: api,
+                                CONF_ACCESS_TOKEN: token,
+                            },
+                        )
                     return self.async_create_entry(
                         title=name, data={CONF_SITE_ID: site, CONF_API_KEY: api}
                     )
```

Comment:
I can make this a single entry creation. I also am in the process of changing it to a username/password instead of extracting the cookie from a browser session.